src/swarm-lang/Swarm/Language/Capability.hs

```diff
@@ -181,20 +64,32 @@ data Capability
     CRecord
   | -- | Debug capability.
     CDebug
-  | -- | Capability to handle keyboard input.
-    CHandleinput
-  | -- | Capability to make other robots halt.
-    CHalt
   | -- | Capability to handle recursive types.
     CRectype
   | -- | God-like capabilities.  For e.g. commands intended only for
     --   checking challenge mode win conditions, and not for use by
     --   players.
     CGod
-  deriving (Eq, Ord, Show, Read, Enum, Bounded, Generic, Hashable, Data, FromJSONKey, ToJSONKey)
+  deriving (Eq, Ord, Show, Generic, Hashable, Data, FromJSONKey, ToJSONKey)
+  deriving (Enum, Bounded) via (FiniteEnumeration Capability)
 
+-- | Get the name of the capability for use in UI and YAML.
 capabilityName :: Capability -> Text
-capabilityName = from @String . map toLower . drop 1 . show
+capabilityName = \case
+  CExecute con -> case con of
+    Neg -> "neg"
+    Not -> "not"
+    _ -> syntax $ constInfo con
```

Comment:
Are we sure `syntax . constInfo` always returns only lowercase?

Reply:
Is that not the same code used to normally parse `Const`? The implementation does use `tShow` so we can be about as sure as possible without writing a newtype.